### PR TITLE
Add changes to validation section

### PR DESCRIPTION
**Pull request by dasuni-30**
### Related Issues
- None

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/validation/api/validation-config.ts [hunk 1]
```diff
@@ -205,3 +218,60 @@ const prepareValidationConfigData = (values: ValidationFormInterface): Validatio
...
+                        value: values.minLength ? values.minLength : "3"
+                    },
+                    {
+                        key: "max.length",
+                        value: values.maxLength ? values.maxLength : "50"
+                    }
+                ],
+                validator: "LengthValidator"
+            }
+        );
```
Comment (pavinduLakshan) [inline apps/console/src/features/validation/api/validation-config.ts:270]: We can use a single `push` to append both rules to the array, can't we?

eg: 

```
const arr: Array = []
arr.push(obj1, obj2,obj3)
console.log(arr)
> [obj1,obj2,obj3]
```

File: apps/console/src/features/validation/api/validation-config.ts [hunk 2]
```diff
@@ -205,3 +218,60 @@ const prepareValidationConfigData = (values: ValidationFormInterface): Validatio
         rules: rules
     };
 };
+
+const prepareUsernameValidationConfigData = (values: ValidationFormInterface): ValidationDataInterface => {
+ 
+    if(values.field=="password"){
+        return;
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/validation/api/validation-config.ts:226]:
```suggestion
```

File: apps/console/src/features/validation/api/validation-config.ts (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by c2507cb1bbad939ec519a8012f3bc2a1abf8cc73.

File: apps/console/src/features/validation/api/validation-config.ts [hunk 3]
```diff
@@ -44,12 +44,21 @@ const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
...
+): Promise<ValidationDataInterface[]> => {
+    const configArray: ValidationDataInterface[] = [
+        passwordData,
+        usernameData,
+        preparePasswordValidationConfigData(formData),
+        prepareUsernameValidationConfigData(formData)
     ];
 
+    const config: ValidationDataInterface[] = configArray.filter((item: ValidationDataInterface) =>
+        item != null);
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by c2507cb1bbad939ec519a8012f3bc2a1abf8cc73.

File: apps/console/src/features/validation/api/validation-config.ts (hunk 2 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by c2507cb1bbad939ec519a8012f3bc2a1abf8cc73.

File: apps/console/src/features/validation/api/validation-config.ts [hunk 4]
```diff
@@ -205,3 +218,57 @@ const prepareValidationConfigData = (values: ValidationFormInterface): Validatio
         rules: rules
     };
 };
+
+const prepareUsernameValidationConfigData = (values: ValidationFormInterface): ValidationDataInterface => {
+ 
+    if(values.field=="password"){
```
Comment (brionmario) [inline apps/console/src/features/validation/api/validation-config.ts:224]:
```suggestion
    if(values.field === "password"){
```

File: apps/console/src/features/validation/api/validation-config.ts [hunk 5]
```diff
@@ -205,3 +218,57 @@ const prepareValidationConfigData = (values: ValidationFormInterface): Validatio
...
+        return;
+    }
+
+    const rules: ValidationConfInterface[] = [];
+
+    if (values.enableValidator=="false"){
```
Comment (brionmario) [inline apps/console/src/features/validation/api/validation-config.ts:230]:
```suggestion
    if (values.enableValidator === "false"){
```

File: apps/console/src/features/validation/api/validation-config.ts [hunk 6]
```diff
@@ -44,12 +44,21 @@ const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
...
+): Promise<ValidationDataInterface[]> => {
+    const configArray: ValidationDataInterface[] = [
+        passwordData,
+        usernameData,
+        preparePasswordValidationConfigData(formData),
+        prepareUsernameValidationConfigData(formData)
     ];
 
+    const config: ValidationDataInterface[] = configArray.filter((item:
+        ValidationDataInterface) => item != null);
```
Comment (thivi) [inline apps/console/src/features/validation/api/validation-config.ts:60]:
```suggestion
    const config: ValidationDataInterface[] = configArray.filter(
        (item: ValidationDataInterface) => item != null);
```

File: apps/console/src/features/validation/api/validation-config.ts [hunk 7]
```diff
@@ -116,7 +125,11 @@ export const useValidationConfigData = <Data = ValidationDataInterface[], Error
     };
 };
 
-const prepareValidationConfigData = (values: ValidationFormInterface): ValidationDataInterface => {
+const preparePasswordValidationConfigData = (values: ValidationFormInterface): ValidationDataInterface => {
+
```
Comment (thivi) [inline apps/console/src/features/validation/api/validation-config.ts:129]: Shall we remove this empty line?

File: apps/console/src/features/validation/api/validation-config.ts (hunk 4 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by a40270675afcfdb91efa3b03d3239d836fd81048.

File: apps/console/src/features/validation/api/validation-config.ts (hunk 6 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by a40270675afcfdb91efa3b03d3239d836fd81048.

File: apps/console/src/features/validation/api/validation-config.ts (hunk 7 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts:129]: Discussed to keep this line for better readability.

File: apps/console/src/features/validation/api/validation-config.ts (hunk 5 above)
Comment (dasuni-30) [reply apps/console/src/features/validation/api/validation-config.ts]: Fixed by 07f527c00c19455576ae8b0a895011d87b0e2aad.

**Review by thivi — APPROVED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Review by kayathiri4 — APPROVED** [review]

**Pull request merged**